samples/sample_encode/src/pipeline_encode.cpp

```diff
@@ -2408,12 +2440,31 @@ mfxStatus CEncodingPipeline::LoadNextFrame(mfxFrameSurface1* pSurf)
         }
     }
     // frameorder required for reflist, dbp, and decrefpicmarking operations
-    if (pSurf) pSurf->Data.FrameOrder = m_nFramesRead;
+    if (pSurf)
+        pSurf->Data.FrameOrder = m_mfxEncParams.mfx.EncodedOrder ? m_QPFileReader.GetCurrentDisplayOrder() : m_nFramesRead;
     m_nFramesRead++;
 
     return sts;
 }
 
+mfxStatus CEncodingPipeline::LoadNextControl(mfxEncodeCtrlWrap*& pCtrl, mfxU32 encSurfIdx)
+{
+    try
+    {
+        if (!pCtrl) return MFX_ERR_NULL_PTR;
+        pCtrl = &m_EncCtrls[encSurfIdx];
+        pCtrl->QP = m_QPFileReader.GetCurrentQP();
+        pCtrl->FrameType = m_QPFileReader.GetCurrentFrameType();
+        m_QPFileReader.NextFrame();
+    }
+    catch (std::exception& /*e*/)
```

Comment:
same question about exception

Reply:
common case

Reply:
then remove. no need to introduce unnecessary exception handling at this level